pkgs/build-support/libredirect/default.nix

```diff
@@ -1,14 +1,15 @@
 { stdenv, lib, coreutils }:
 
-stdenv.mkDerivation {
-  name = "libredirect-0";
+stdenv.mkDerivation rec {
+  pname = "libredirect";
+  version = "0";
 
   unpackPhase = ''
     cp ${./libredirect.c} libredirect.c
     cp ${./test.c} test.c
   '';
 
-  libName = "libredirect" + stdenv.targetPlatform.extensions.sharedLibrary;
+  libName = "${pname}${stdenv.targetPlatform.extensions.sharedLibrary}";
```

Comment:
This probably should not depend on `pname`. Package gets name from the library, not the other way around.